VEP-6: Vitess Release Cadence to 6 monthly cycle (#12)
vitessio/enhancements: veps/vep-6.md

This VEP proposes a change to how often releases are made thus replacing all of VEP-5.

All other sections of VEP-1 will continue to be in force without any change.

## Versioning and Release Cadence

The proposed release cycle is best described as _Monotonic Versioning_, using the terminology `COMPATIBILITY.RELEASE` from the [Monotonic Versioning Manifesto](http://blog.appliedcompscilab.com/monotonic_versioning_manifesto/):

1. The `COMPATIBILITY` number will be incremented every 6 months. For example, the next major release will have a `COMPATIBILITY` number of `22.0`.
2. Each `COMPATIBILITY` number will be supported by the Open Source Vitess community for 12 months after the initial release.
3. Patch releases will be created as needed. Patch releases will follow SemVer. The numbering scheme will be `COMPATIBILITY.PATCH_NUMBER`. For example, the first patch release on `22.0`, if there is one, will be `22.0.1`. `PATCH_NUMBER` will be monotonically increasing.

> 4. Events that trigger patch releases are major regressions in functionality or performance, and CVEs.

Reviewer: Shouldn't a critical bug in newly introduced functionality also trigger a patch release? Or is it implied by `major regressions in functionality`

Reviewer 2: Yeah, IMO this is slightly better:
```
4. Events that trigger patch releases are critical bugs, major regressions in functionality or performance, and CVEs.
```

Author: I will update it

> # # VEP-6 Vitess Release cadence

Reviewer 3: Why not keep calling this Cycle instead of Cadence? And let's make sure to match case on this headline and the `Title` on line 5.

> # # VEP-6 Vitess Release cadence

Reviewer 3: ```suggestion
# # VEP-6 Vitess Release Cycle
```

```
VEP: 6

> Title: Vitess Release Cadence

Reviewer 3: ```suggestion
Title: Vitess Release Cycle
```

Author: Harshit Gangal
Reviewers: 
Status: 
Created: 2024-11-02
```

## Abstract

[VEP-1](https://github.com/vitessio/enhancements/blob/main/veps/vep-1.md) established a support lifecycle for Vitess. [VEP-2](https://github.com/vitessio/enhancements/blob/main/veps/vep-2.md) later amended parts of it.

> [VEP-5](https://github.com/vitessio/enhancements/blob/main/veps/vep-5.md) modified the yearly major release to 3 and 1 year support for each major release, thus replaces the Support lifecycle section in VEP-1, and all of VEP-2.

Reviewer 3: ```suggestion
[VEP-5](https://github.com/vitessio/enhancements/blob/main/veps/vep-5.md) changed the frequency of major releases to 3 per year and established a 1 year support cycle for each major release, thus replacing the Support Lifecycle section in VEP-1, and all of VEP-2.
```

> This VEP suggests a change to how often releases are made and its support lifecycle thus replacing all of VEP-5.

Reviewer 3: ```suggestion
This VEP proposes a change to how often releases are made thus replacing all of VEP-5.
```

All other sections of VEP-1 will continue to be in force without any change.

## Versioning and Release Cadence

> The proposed release cycle is best described as _Monotonic Versioning_ with a release cadence similar to Kubernetes.

Reviewer 3: similar to Kubernetes is no longer correct. They are still doing 3 a year. It may be ok to delete this entire line. Or just keep the first part of it.

Author: done

# # VEP-6 Vitess Release cadence

```
VEP: 5
Title: Vitess Release Cadence

> Author: Andres Taylor

Reviewer 4: Shouldn't it be @harshit-gangal?
